kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/CompactCommand.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.cli.commands;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+import org.apache.crunch.PipelineResult;
+import org.kitesdk.data.View;
+import org.kitesdk.tools.CompactionTask;
+import org.slf4j.Logger;
+
+import static org.apache.avro.generic.GenericData.Record;
+
+@Parameters(commandDescription="Compact all or part of a dataset")
```

Comment:
Does this command handle the case where there is a writer writing to the same partitions that are being compacted?